nixos/modules/services/networking/soju.nix

```diff
@@ -0,0 +1,126 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.soju;
+  listenCfg = concatMapStringsSep "\n" (l: "listen ${l}") cfg.listen;
+  tlsCfg = optionalString (cfg.tlsCertificate != null)
+    "tls ${cfg.tlsCertificate} ${cfg.tlsCertificateKey}";
+  logCfg = optionalString (cfg.logDirectory != null)
+    "log fs ${cfg.logDirectory}";
+
+  configFile = pkgs.writeText "soju.conf" ''
+    ${listenCfg}
+    hostname ${cfg.hostName}
+    ${tlsCfg}
+    db sqlite3 ${cfg.databasePath}
```

Comment:
Presumably this service will crash immediately after it starts if the sysadmin provides any value outside of `/var/lib/soju` for `databasePath`.

You might consider renaming `databasePath` to `stateDir`, setting a default to `/var/lib/soju`, and then conditionally provisioning the directory via `StateDirectory` (as you do below) depending on whether the default value is set or not.

Check out [this](https://github.com/NixOS/nixpkgs/pull/93001) PR to see more specifically what I mean.

Reply:
Good point. I was just naively copying all of soju's options. How about we just make this non-configurable? I can't imagine a use case for changing the database path.

Reply:
That certainly simplifies things :+1: